File: common1/js_cpp/basic_pros.js
(function () {
  let a = 10; // 변수 할당 방법

  function sum(a, b) { // 함수 선언 방법
    if (a < 0) { // 조건문 작성 방법
      a *= -1;
    }
    return (a+b);
  }

  console.log(sum(5,5)); // 출력 방법 결과값: 10

  for (let i=0; i<10; i++) { // 기본적인 반복문 작성 방법
    console.log(i);
  }

  const person = {
    name: 'kmin',
    age: 27,
    country: 'korea',
  };

  for (const value in person) { // 객체를 순회하는 반복문
    console.log(value); // kmin 27 korea
  }

  const arr = [1, "hello", false];
  /* 반복 가능한 객체
  (Array, Map, Set, String, TypedArray, arguments)
  를 순회하는 반복문 */
  for (const value of arr) {
    console.log(value);
  }
})();

File: common1/js_cpp/basic_pros.cpp
#include <iostream>

struct s_person
{
    int age;
    std::string phone_number;
    std::string name;
};

void print_person(const s_person *person)
{
    std::cout << person->age << std::endl;
    std::cout << person->phone_number << std::endl;
    std::cout << person->name << std::endl;
}

int a = 10; // 변수 할당 방법

int sum(int a, int b) // 함수 선언 방법
{
    if (a < 0)
    { // 조건문 작성 방법
        a *= -1;
    }
    return (a + b);
}

int main()
{
    std::cout << sum(5, 5) << std::endl; // 출력 방법

    for (int i = 0; i < 10; i++)
    { // 반복문 작성 방법
        std::cout << i << std::endl;
    }

    s_person *person;
    if (!(person = (s_person *)malloc(sizeof(s_person))))
    {
        return (0);
    }
    person->age = 10;
    person->phone_number = "010-1234-5678";
    person->name = "kmin";
    print_person(person);
    free(person);

    return (0);
}
